pumpkin/src/client/player_packet.rs

```diff
@@ -400,6 +400,10 @@ impl Client {
     }
 
     pub fn handle_set_creative_slot(&mut self, _server: &mut Server, packet: SSetCreativeSlot) {
+        let gamemode = self.player.as_ref().unwrap().gamemode;
+        if gamemode != GameMode::Creative {
+            self.kick("CHEATER")
```

Comment:
Im also not sure about the kick message, I generally like to use a "more professional", I want to check for vanilla impossible actions at multiple places. I prefer to use "Invalid  ... action". In this case we could use. "Invalid Gamemode". A modder/cheat developer will properbly understand what he did wrong

Reply:
fixed :)